metals/src/main/scala/scala/meta/internal/builds/NewProjectProvider.scala

```diff
@@ -0,0 +1,358 @@
+                .creationFailed(template, parent.toString())
+            )
+          }
+      }
+  }
+
+  private def askForWindow(projectPath: AbsolutePath): Future[Unit] = {
+    def openWindow(newWindow: Boolean) = {
+      val params = MetalsOpenWindowParams(
+        projectPath.toURI.toString(),
+        new java.lang.Boolean(newWindow)
+      )
+      val command = new ExecuteCommandParams(
+        ClientCommands.OpenWindow.id,
+        List[Object](
+          params.toJsonObject
+        ).asJava
+      )
+      client.metalsExecuteClientCommand(command)
+    }
+
+    if (config.isOpenNewWindowProvider()) {
+      client
+        .showMessageRequest(NewScalaProject.askForNewWindowParams())
+        .asScala
+        .map {
+          case msg if msg == NewScalaProject.no =>
+            openWindow(newWindow = false)
+          case msg if msg == NewScalaProject.yes =>
+            openWindow(newWindow = true)
+          case _ =>
+        }
+    } else {
+      Future.successful {
+        client.showMessage(NewScalaProject.newProjectCreated(projectPath))
+      }
+    }
+  }
+
+  private def askForTemplate(
+      templates: Seq[MetalsQuickPickItem]
+  ): Future[Option[MetalsQuickPickItem]] = {
+    client
+      .metalsQuickPick(
+        MetalsQuickPickParams(
+          templates.asJava,
+          placeHolder = NewScalaProject.selectTheTemplate
+        )
+      )
+      .asScala
+      .flatMap {
+        case kind if kind.itemId == NewProjectProvider.more.id =>
+          askForTemplate(allTemplatesFromWeb)
+        case kind if kind.itemId == NewProjectProvider.back.id =>
+          askForTemplate(NewProjectProvider.curatedTemplates(icons))
+        case kind if kind.itemId == NewProjectProvider.custom.id =>
+          askForName("", NewScalaProject.enterG8Template)
+            .mapOptionInside { g8Path =>
+              MetalsQuickPickItem(
+                nameFromPath(g8Path),
+                g8Path,
+                NewProjectProvider.custom.description
+              )
+            }
+        case kind if !kind.cancelled =>
+          Future.successful(
+            templates
+              .find(_.id == kind.itemId)
+          )
+        case _ => Future.successful(None)
+      }
+  }
+
+  private def askForName(
+      default: String,
+      prompt: String
+  ): Future[Option[String]] = {
+    if (config.isInputBoxEnabled()) {
+      client
+        .metalsInputBox(
+          MetalsInputBoxParams(
+            prompt = prompt,
+            value = default
+          )
+        )
+        .asScala
+        .flatMap {
+          case name if !name.cancelled && name.value.nonEmpty =>
+            Future.successful(Some(name.value))
+          case name if name.cancelled =>
+            Future.successful(None)
+          // reask if empty
+          case _ => askForName(default, prompt)
+        }
+    } else {
+      Future.successful(Some(default))
+    }
+  }
+
+  private def constructPath(
```

Comment:
'askForPath' (or sth similar to good convention askForName / askForTemplate)
Nice recursion :)

Reply:
Changed the name!